### PR TITLE
[AutoScheduler] Allow device specification for AutoScheduler Runners.

### PR DESCRIPTION
**Pull request by jwfromm**
Currently, all measurements in the autoscheduler are run on device 0. Although this is fine for most cases, on some platforms with multiple gpus, it may be important to run on a specific device. This small PR adds an optional `device` argument to the various `Runner` classes in `auto_scheduler/measure` to address this limitation.

### CODE REVIEW

**Comment by jwfromm** [comment]
@tkonolige can you take a look at this PR?

File: python/tvm/auto_scheduler/measure.py [hunk 1]
```diff
@@ -1257,6 +1276,8 @@ def rpc_runner_run(
         This is only has effect on CPU task.
     verbose: int = 1
         Verbosity level. 0 for silent, 1 to output information during program measuring.
+    device: int = 0
+        Which device to run on if multiople are available.
```
Comment (tkonolige) [inline python/tvm/auto_scheduler/measure.py:1280]:
```suggestion
        Which device to run on if multiple are available.
```

File: tests/python/relay/test_auto_scheduler_tuning.py [hunk 2]
```diff
@@ -36,7 +36,7 @@ def tune_network(network, target):
         log_file = fp.name
 
         # Tuning
-        measure_ctx = auto_scheduler.LocalRPCMeasureContext(timeout=60)
+        measure_ctx = auto_scheduler.LocalRPCMeasureContext(timeout=60, device=0)
```
[the remark this answers is not in the document]
Comment (jwfromm) [reply tests/python/relay/test_auto_scheduler_tuning.py:39]: Yeah I dont think it would work on any of the CI targets unfortunately.

**Comment by jwfromm** [comment]
@masahi can you also check to make sure this all looks ok?

**Review by masahi — APPROVED** [review]

**Pull request merged**